src/modules/phoneNumber/parsePhoneNumber.ts

```diff
@@ -0,0 +1,49 @@
+import { withErrorBoundary } from '../../common/errorBoundary';
+import { PHONE_FORMATTER_MAPPER } from './data/phoneFormatterMapper';
+import formatPhoneNumber from './formatPhoneNumber';
+import { detectCountryCodeFromDialCode, cleanPhoneNumber } from './utils';
+
+interface PhoneInfo {
+  countryCode: string;
+  dialCode: string;
+  formattedPhoneNumber: string;
+  formatTemplate: string;
+}
+
+const parsePhoneNumber = (
+  phoneNumber: string,
+  country?: string,
+): PhoneInfo | null => {
+  const countryCode =
+    country in Object.keys(PHONE_FORMATTER_MAPPER)
```

Comment:
won't this condition always return false ? `in` operator will check for indices in array. 
We can avoid Object.keys here